libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/accounts/MapboxNavigationAccounts.kt

```diff
@@ -0,0 +1,50 @@
+package com.mapbox.services.android.navigation.v5.internal.accounts
+
+import android.content.Context
+import android.text.format.DateUtils
+import com.mapbox.android.accounts.navigation.sku.v1.MauSku
+import com.mapbox.android.accounts.navigation.sku.v1.SkuGenerator
+import com.mapbox.android.accounts.navigation.sku.v1.TripsSku
+import com.mapbox.android.accounts.v1.AccountsConstants.MAPBOX_SHARED_PREFERENCES
+
+internal class MapboxNavigationAccounts private constructor() {
+
+    companion object {
+        private const val MAU_TIMER_EXPIRE_THRESHOLD = 1
+        private const val TRIPS_TIMER_EXPIRE_THRESHOLD = 2
+        private const val TRIPS_REQUEST_COUNT_THRESHOLD = 5
+        private const val TIMER_EXPIRE_AFTER = DateUtils.HOUR_IN_MILLIS / 1000
+        private var skuGenerator: SkuGenerator = DisabledSku()
+        private var INSTANCE: MapboxNavigationAccounts? = null
+
+        @JvmStatic
+        fun getInstance(context: Context): MapboxNavigationAccounts =
+                INSTANCE ?: synchronized(this) {
+                    MapboxNavigationAccounts().also { mapboxNavigationAccount ->
+                        INSTANCE = mapboxNavigationAccount
+                        // TODO uncomment when ready to release as a part of 1.0
```

Comment:
Should we remove this commented out code now as we're going to merge this PR into `master` and start testing on the field?